### DISCUSSION TITLE
Github App and scopes (NOT OAuth apps)

### DISCUSSION THREAD

**Discussion by nichita-pasecinic** [community/community#42701] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi I've got a question on using Github App with tokens that have `scope`.

I'm using a Github App (with all permissions set) in order to generate an access token (with default expiration).
The response I'm receiving is something like so:
```js
{
  access_token: 'ghu_asdf',
  expires_in: 28800,
  refresh_token: 'ghr_asdf',
  refresh_token_expires_in: 15724799,
  token_type: 'bearer',
  scope: ''
}
```

Note that the `scope` is empty, I have some third party integrations (e.g.: Vault) that requires a scope of `read:org` ([docs](https://developer.hashicorp.com/vault/docs/auth/github))
![image](https://user-images.githubusercontent.com/98735377/209660326-c0d5107f-9826-4cf2-8fb8-e7583f776000.png)

Is there any possible workaround to generate an access token via a Github App that would have the `scope` field set ?

Thanks!

**Comment by anup-kodlekere** [comment]
Hey @nichita-pasecinic, I'm in need of something similar. Did you manage to find an answer for this?

**Comment by nichita-pasecinic** [comment]
@anup-kodlekere I believe there is not real way to achieve this via GithubApp, you'll need to use OAuth Flow for this https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/differences-between-github-apps-and-oauth-apps#requesting-permission-levels-for-resources

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @nichita-pasecinic ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. I've gone ahead and moved it for you. Good luck!
